Sources/WhisperKit/Core/AudioProcessor.swift

```diff
@@ -585,6 +593,22 @@ public extension AudioProcessor {
 
         audioEngine = try setupEngine(inputDeviceID: inputDeviceID)
 
+        // Set the callback
+        audioBufferCallback = callback
+        
+        lastInputDevice = inputDeviceID
+    }
+    
+    func resumeRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)? = nil) throws {
+        try? setupAudioSessionForDevice()
+        
+        if inputDeviceID == lastInputDevice{
+            try audioEngine?.start()
+        } else {
+            audioEngine = try setupEngine(inputDeviceID: inputDeviceID)
+        }
+        
+
         // Set the callback
         audioBufferCallback = callback
```

Comment:
Looking at it again, we probably won't want this to be overwritten when resuming if callback is nil and audioBufferCallback isn't nil

```suggestion
        // Set the callback only if the provided callback is not nil
        if let callback = callback {
            audioBufferCallback = callback
        }
```